extensions/rvdatabinding/src/main/java/com/google/android/agera/rvdatabinding/DataBindingRepositoryPresenterCompiler.java

```diff
@@ -49,12 +48,10 @@
 final class DataBindingRepositoryPresenterCompiler
     implements DBRPMain, RPLayout, RPTypedCollectionCompile {
   @NonNull
-  private static final Function NO_ID = staticFunction(-1);
-  @NonNull
   private final SparseArray<Object> handlers;
   private Function<Object, Integer> layoutFactory;
-  private Function<Object, Integer> itemId = NO_ID;
-  private Function<Object, Integer> collectionId = NO_ID;
+  private Function<Object, Integer> itemId = staticFunction(0);
+  private int collectionId = -1;
```

Comment:
umm... you're using both 0 and -1? :D

BR IDs seem to start from 0 where 0 is always `_all`. I don't know what this special ID does. I guess -1 is the best bet (and simply use `!= -1` to detect a valid ID)